torch/_inductor/lowering.py

```diff
@@ -413,38 +413,54 @@ def inner_fn(index):
     return inner
 
 
-def make_foreach_pointwise(aten_fn, pw_fn):
+def make_foreach_pointwise(pw_fn, allow_alpha=False):
     def inner(*inputs: List[List[TensorBox]], alpha=1):
-        def is_dynamic(t):
-            return any(x.free_symbols for x in t.data.get_size())
+        def is_dynamic(*args):
+            return any(isinstance(t, TensorBox) and any(x.free_symbols for x in t.data.get_size()) for t in args)
+
+        def has_type_promotion(*args):
+            if len(args) < 2:
```

Comment:
can you add specific tests for sqrt? even though sqrt is a unary function, it type promotes - it will have fp outputs for integral inputs

Reply:
I didn't see tests for int -> float for sqrt, did I miss them?